web/ui/src/common/miscUtils.js

```diff
@@ -264,6 +264,35 @@
 
                 }
                 return null;
+            },
+            
+            // Forms the name displayed for a public endpoint.
+            publicEndpointURL: function($scope, publicEndpoint){
```

Comment:
passing `$scope` into a util is leaky. In this case, all that is needed is `$scope.defaultHostAlias`, so pass that in instead.